subsys/net/ip/net_shell.c

```diff
@@ -3532,6 +3532,173 @@ static int cmd_net_ipv6(const struct shell *shell, size_t argc, char *argv[])
 	return 0;
 }
 
+static int cmd_net_ip6_add(const struct shell *shell, size_t argc, char *argv[])
+{
+#if defined(CONFIG_NET_NATIVE_IPV6)
+	struct net_if *iface = NULL;
+	int idx;
+	struct in6_addr addr;
+
+	if (argv[1]) {
+		idx = get_iface_idx(shell, argv[1]);
+		if (idx < 0) {
+			return -ENOEXEC;
+		}
+
+		iface = net_if_get_by_index(idx);
+		if (!iface) {
+			PR_WARNING("No such interface in index %d\n", idx);
+			return -ENOEXEC;
+		}
+	if (argv[2]) {
+		if (net_addr_pton(AF_INET6, argv[2], &addr)) {
+			PR_ERROR("Invalid address: %s\n", argv[2]);
+			return -EINVAL;
+			}
+		net_if_ipv6_addr_add(iface, &addr, NET_ADDR_MANUAL, 0);
+
+		return 0;
+		}
+		PR_ERROR("Need address and prefix length(optional)\n");
+		return -1;
+	}
+#endif /* CONFIG_NET_NATIVE_IPV6 */
+}
+
+static int cmd_net_ip6_del(const struct shell *shell, size_t argc, char *argv[])
+{
+#if defined(CONFIG_NET_NATIVE_IPV6)
+	struct net_if *iface = NULL;
+	int idx;
+	struct in6_addr addr;
+
+	if (argv[1]) {
+		idx = get_iface_idx(shell, argv[1]);
+		if (idx < 0) {
+			return -ENOEXEC;
+		}
+
+		iface = net_if_get_by_index(idx);
+		if (!iface) {
+			PR_WARNING("No such interface in index %d\n", idx);
+			return -ENOEXEC;
+		}
+		if (argv[2]) {
+			if (net_addr_pton(AF_INET6, argv[2], &addr)) {
+				PR_ERROR("Invalid address: %s\n", argv[2]);
+				return -EINVAL;
+				}
+			if (!net_if_ipv6_addr_rm(iface, &addr)) {
+				PR_ERROR("Failed to delete %s\n", argv[2]);
+				return -1;
+			}
+			return 0;
+		}
+		PR_ERROR("Need address that needs to be deleted\n");
+		return -1;
+	}
+#endif /* CONFIG_NET_NATIVE_IPV6 */
+	return 0;
+}
+
+static int cmd_net_ip6_route_add(const struct shell *shell, size_t argc, char *argv[])
+{
+#if defined(CONFIG_NET_NATIVE_IPV6)
+	struct net_if *iface = NULL;
+	int idx;
+	int prefix_len = 0;
+	struct net_route_entry *route;
+	struct in6_addr gw = {0};
+	struct in6_addr prefix = { 0 };
+
+	if (argv[1]) {
+		idx = get_iface_idx(shell, argv[1]);
+		if (idx < 0) {
+			return -ENOEXEC;
+		}
+
+		iface = net_if_get_by_index(idx);
+		if (!iface) {
+			PR_WARNING("No such interface in index %d\n", idx);
+			return -ENOEXEC;
+		}
+
+		if (argv[2]) {
```

Comment:
I don't think that in general it's a right way to check if an argument is present. If for example `argc == 1`, `argv[2]` could be anything (out of bound access).

Reply:
removed argv checks and added arg count check.